modules/template/elements/BaseElementVariable.php

```diff
@@ -8,89 +8,29 @@
 
 namespace humhub\modules\custom_pages\modules\template\elements;
 
-use humhub\modules\custom_pages\modules\template\interfaces\TemplateElementContentIterable;
-
 class BaseElementVariable
 {
-    private array $options = [];
-
-    private BaseElementContent $elementContent;
-
-    public function __construct(BaseElementContent $elementContent, string $mode = 'edit')
-    {
-        $this->elementContent = $elementContent;
-        $this->options['mode'] = $mode;
-    }
+    protected BaseElementContent $elementContent;
 
-    public function getLabel()
-    {
-        return $this->elementContent->getLabel();
-    }
-
-    public function isEditMode(): bool
-    {
-        return isset($this->options['mode']) && $this->options['mode'] === 'edit';
-    }
+    public int $elementContentId;
 
-    public function getEmpty()
-    {
-        return $this->elementContent->isEmpty();
-    }
+    public bool $empty;
 
-    public function getContent()
+    public function __construct(BaseElementContent $elementContent)
     {
-        return $this->elementContent;
+        $this->elementContent = $elementContent;
+        $this->elementContentId = $elementContent->id ?? 0;
+        $this->empty = $elementContent->isEmpty();
     }
 
-    public function render(): string
+    public static function instance(BaseElementContent $elementContent): static
```

Comment:
Whats the benefit here to introduce a second static constructor?

Reply:
@luke- I have implemented the `instance` method in order to use code like this:

- `BaseRecordElementVariable::instance($elementContent)->setRecord($record)`

Reply:
`return (new BaseRecordElementVariable($elementContent))->setRecord($record)` should work too? 
But if you prefer `::instance` we can stay with that solution.